pkg/controller/mizar/mizar-service-controller.go

```diff
@@ -213,115 +214,143 @@ func (c *MizarServiceController) syncService(eventKeyWithType KeyWithEventType)
 
 	response := c.grpcAdaptor.CreateService(c.grpcHost, msg)
 	code := response.Code
 	ip := response.Message
-	klog.Infof("Assigned ip by mizar is %v", ip)
+	klog.V(4).Infof("Assigned ip by mizar is %v", ip)
 
 	switch code {
 	case CodeType_OK:
-		klog.Infof("Mizar handled service creation successfully: %s", key)
-		if beginsWithKubernetes(service.Name) {
-			endpoint, err := c.kubeClientset.CoreV1().EndpointsWithMultiTenancy("default", "system").Get(kubernetesSvcDefaultName, metav1.GetOptions{})
+		klog.V(4).Infof("Mizar handled service creation successfully: %s", key)
+		if beginsWithKubernetes(service.Name) {	// Is this logic expected?
+			kubernetesEndpoint, err := c.kubeClientset.CoreV1().EndpointsWithMultiTenancy(metav1.NamespaceDefault, metav1.TenantSystem).Get(kubernetesSvcDefaultName, metav1.GetOptions{})
 			if err != nil {
-				klog.Errorf("The get endpoint failed to get: %v", endpoint)
+				klog.Errorf("Failed to get kubernetes endpoint: %v. Error: %v", kubernetesEndpoint, err)
 				return err
 			}
-			message := convertToServiceEndpointMessage(service.Name, service.Namespace, service.Tenant, endpoint, service)
+			message := convertToServiceEndpointMessage(service.Name, service.Namespace, service.Tenant, kubernetesEndpoint, service)
 			resp := c.grpcAdaptor.CreateServiceEndpoint(c.grpcHost, message)
 			returnCode := resp.Code
 			context := resp.Message
 			switch returnCode {
 			case CodeType_OK:
-				klog.Infof("Mizar handled kubernetes network service endpoint successfully: %s", context)
+				klog.V(4).Infof("Mizar handled kubernetes network service endpoint successfully: %s", context)
 			case CodeType_TEMP_ERROR:
 				klog.Warningf("Mizar hit temporary error for kubernetes network service endpoint: %s", context)
+				// Question - does this need to be retry?
 			case CodeType_PERM_ERROR:
 				klog.Errorf("Mizar hit permanent error for kubernetes network service endpoint %s", context)
 			}
 		}
 	case CodeType_TEMP_ERROR:
 		klog.Warningf("Mizar hit temporary error for service creation for service: %s", key)
-		c.queue.AddRateLimited(eventKeyWithType)
 		return errors.New("Service creation failed on mizar side, will try again.....")
 	case CodeType_PERM_ERROR:
 		klog.Errorf("Mizar hit permanent error for service creation for service: %s", key)
 		return errors.New("Service creation failed permanently on mizar side")
 	}
 
+	// update service cluster ip with ip from mizar
 	if len(service.Spec.ClusterIP) == 0 {
 		svcToUpdate := service.DeepCopy()
 		svcToUpdate.Spec.ClusterIP = ip
-		svcUpdated, err := c.kubeClientset.CoreV1().ServicesWithMultiTenancy(service.Namespace, service.Tenant).Update(svcToUpdate)
+		_, err := c.kubeClientset.CoreV1().ServicesWithMultiTenancy(service.Namespace, service.Tenant).Update(svcToUpdate)
 		if err != nil {
-			klog.Errorf("The following service failed to update: %v", svcUpdated)
+			klog.Errorf("Failed to update service %v/%v/%v: %v", service.Tenant, service.Namespace, service.Name, err)
```

Comment:
any cleanup from mizar is needed with the failure case here ?

Reply:
Actually Mizar does not change service up so this part of code I don't think it will have chance to run. Keep it there until we have full understanding of how all services works together.